hashintel/hash
# Properly import JS modules

## 🌟 What is the purpose of this PR?

This PR changes JS packages file from scripts to modules.
In addition to the correctness modules bring, we'll be able to add new JS files without modifying Rust code.

## ⚠️ Known issues

Dynamic imports and more generally top level await probably doesn't work.

## 🔍 What does this change?

- Load and evaluate packages as modules and not scripts.
- Convert packages to use imports instead of wrapping functionally in a function and pass dependencies as function arguments.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201959586244685/1202018802226446/f) _(internal)_

## 🛡 What tests cover this?

- Tests with a JS runner
- The `boids_3d` example tests that the `hash_stdlib` is available

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2.  Try `cargo test`
3.  Confirm that tests are passing
